CODING_GUIDLINES.md

```diff
@@ -0,0 +1,117 @@
+
+⚠️ YOU SHOULD NOT strongly consider not doing this. If not following this advice, please comment code with why and proactively begin a discussion as part of the PR process.
+
+✔️ YOU MAY consider this advice if appropriate to your situation. Other team members may comment on this as part of PR review, but there is no need to be proactive.
+
+## Development Environment
+
+☑️ YOU SHOULD use VSCode and install the following extensions:
+
+- [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
+- [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
+
+This setup will catch most linting/formatting/type errors as you develop.
+
+☑️ YOU SHOULD use the [GitHub CLI](https://cli.github.com/). It has helpful workflows for submitting PRs as well as for checking out other team members PRs.
+
+✔️ YOU MAY use Windows, Linux (including WSL), or OSX. We have team members on all three environments. When modifying any engineering systems please keep cross platform compatability in mind.
+
+## Formatting
+
+✅ DO use [Prettier](https://prettier.io/) to format your code. This will be preformed automatically if using the recommended editor setup. Otherwise `npm run format` will also format code.
+
+## Linting
+
+✅ DO use [ESLint](https://eslint.org/) to check for linting errors your code. This can also be performed by `npm run lint`
+
+✔️ YOU MAY consider adding new lint rules. If you find yourself performing "nits" as part of PR review, consider if it is possible to add a lint rule that will autuomatically catch the error in the future
+
+⚠️ YOU SHOULD NOT disable any lint rules. They are there as guidance and to catch common mistakes. You will find places we disable specific lint rules, but is should be exceptional. If a rule does need to be disabled, prefer disabling a specific line instead of the entire file.
+
+⛔️ DO NOT add any tslint rules. TSLint has been deprecated and is on track to be removed. Always prefer ESLint rules.
+
+## UI Components
+
+☑️ YOU SHOULD write new components using React. We are actively migrating Cosmos Explorer off of Knockout.
+
+☑️ YOU SHOULD use Fluent UI components. These are highly accesible and composable. They allow us to build upon the work of the Fluent team and lead to a lower total cost of ownership for our UI code.
+
+### React
+
+⛔️ DO NOT use inheritance for sharing component behavior. The React documentation covers this topic in detail https://reactjs.org/docs/composition-vs-inheritance.html
+
+☑️ YOU SHOULD use pure functional components when no state is required
+
+✔️ YOU MAY consider using functional (hooks) or class components. The project contains examples of both and neither is strongly preferred right now.
+
+## Libraries
+
+⛔️ DO NOT add new libraries to package.json. Adding libraries has the possibility to bring in code that explodes our bundled size and/or attempts to run NodeJS code in the browser. Consult with project owners first.
```

Comment:
This feels like a `SHOULD NOT` instead of a `MUST NOT`, given that the guidance is to consult with project owners first.

Reply:
Or a `MUST` consult with the project owners before adding a package which is more specific.